jenkins.sh

```
@@ -4,6 +4,12 @@
 : "${JENKINS_HOME:="/var/jenkins_home"}"
 : "${COPY_REFERENCE_FILE_LOG:="${JENKINS_HOME}/copy_reference_file.log"}"
 : "${REF:="/usr/share/jenkins/ref"}"
+
+if ! [ -r "${JENKINS_HOME}" ] && [ -w "${JENKINS_HOME}" ]; then
+    echo INSTALL WARNING: User: "${USER}" missing rw permissions on JENKINS_HOME: "${JENKINS_HOME}";
+    echo "    " Jenkins unable to start. To fix, use chown or setfacl -m u:"${USER}":rwx "${JENKINS_HOME}";
+    exit 1;
```

From a stylistic point of view, there's no need for the trailing semicolon at the end of each line in a script, and the quoting can be simplified to:

```bash
        echo "INSTALL WARNING: User: ${USER} missing rw permissions on JENKINS_HOME: ${JENKINS_HOME}"
        echo "    Jenkins unable to start. To fix, use chown or setfacl -m u:${USER}:rwx ${JENKINS_HOME}"
```

I personally support the (strict) behavior of rejecting such a configuration and failing fast. But if this is too harsh, I could see the other perspective of merely tolerating such a configuration and printing a (soft) warning.

---

Simplified

---

@clintonsteiner big thank you for your patience with this!

> Jenkins unable to start. To fix, use chown or setfacl 

If we don't exit, and we support read-only JENKINS_HOME with writable subdirectories (however strange it might be), we should not say "unable to start".

In my practice very often permission problems are also related to file systems mounted read-only, so "chown/setfacl" does not apply there.

Shall we just drop that second line?

---

Yeah that works for me, I ran into this issue installing it on via unraid template and wanted to offer improved troubleshooting for the next person.

I will drop the second line and squash momentarily

---

Squashed. Thank you everyone! @saper